CONTRIBUTING.md

```diff
@@ -0,0 +1,100 @@
+
+```
+pip install -e .[dev]
+```
+
+When using the flag [`-e`](https://pip.pypa.io/en/stable/cli/pip_install/#install-editable), pip does not copy the package is to `site-packges`, but creates a link to your local repository. So any changes to the sourcecode are directly reflected in the 'installed' package. Installing the optional `[dev]` dependencies makes sure all tools for style checking, testing and building documentation locally are available.
+
+
+## Code style
+SleepECG adheres to [PEP 8](https://www.python.org/dev/peps/pep-0008/), with the following exceptions/specifications:
+- Each source file contains a header listing the authors contributing to that module and mentions the license:
+    ```python
+    # Authors: Firstname Lastname
+    #          Another Name
+    #
+    # License: BSD (3-clause)
+    ```
+- The maximum line length is `92` (instead of `79`).
+    - If readability would suffer from a linebreak, append `# noqa` to the relevant line to disable line length checking.
+- Single quotes are used, except if the string contains an apostrophe.
+- [Type hints](https://www.python.org/dev/peps/pep-0484/) are encouraged.
+- If a container literal, function definition or function call does not fit on one line:
+    - Each argument or item is indented one level further than the function or container name.
+    - The last argument or item has a trailing comma
+    - The closing parenthesis or bracket is indented at the same level as the starting line.
+    ```python
+    # Example
+    def download_file(
+        url: str,
+        target_filepath: Path,
+        checksum: Optional[str] = None,
+        checksum_type: Optional[str] = None,
+        verbose: bool = False,
+    ) -> None:
+    ```
+
+
+## Documentation
+For docstrings, SleepECG mainly follows [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html), with the following exceptions/specifications:
+- The maximum line length is `75`. You can use
+- For parameters that may take multiple types, pipe characters are used instead of the word `or`, like this: `param_name : int | float`.
+- For single return values, only the type is stated (no name).
+- For multiple return values, both a name and a type are stated.
+- Generators are treated similarly, so a type annotation `Iterator[int]` becomes just `int` in the docstring.
+
+
+## pre-commit
+Coding and documentation style are checked via a CI job. To make sure your contribution passes those checks, you can use [`pre-commit`](https://pre-commit.com/). To install the hooks configured in `.pre-commit-config.yml`, run
+
+```
+pre-commit install
+```
+
+inside your local clone. After that, the checks required in the CI job will be run on all staged files when you commit - and abort the commit in case any issues are found.
```

Comment:
```suggestion
inside your local clone. After that, the checks required in the CI job will be run on all staged files when you commit – and abort the commit in case any issues are found (in which case you should fix the found issues and commit again).
```